cortex/utils.py

```diff
@@ -100,79 +101,86 @@ def named_dict(obj):
     if isinstance(obj, dict):
         for key, value in obj.items():
             obj[key] = named_dict(value)
-        return namedtuple('NamedDict', obj.keys())(**obj)
-    elif isinstance(obj, list):
+        return namedtuple("NamedDict", obj.keys())(**obj)
+    if isinstance(obj, list):
         return [named_dict(item) for item in obj]
-    else:
-        return obj
+    return obj
 
 
-def decode_JWT(*args):
+def decode_JWT(*args) -> tuple:
+    # pylint: disable=invalid-name
     """
     thin wrapper around jwt.decode. This function exists for better error handling of the
     jwt exceptions.
     """
-    invalidTokenMsg = 'Your Cortex Token is invalid: '
+    invalid_token_msg = "Your Cortex Token is invalid: "
     try:
-        (header, payload) = decodedJWT = py_jwt.process_jwt(*args)
-        # there are places in the sdk where we try to decode 'any ol token' before sending the token to kong to get verified
-        # therefore, here we have some reasonable checks to make sure that this is a cortex token by checking the JWT keys exist
-        if not payload.get('aud') or not payload.get('sub') or not payload.get('exp'):
-            raise BadTokenException(invalidTokenMsg)
+        (_, payload) = decodedJWT = py_jwt.process_jwt(*args)
+        # there are places in the sdk where we try to decode 'any ol token' before sending the token
+        # to auth to get verified therefore, here we have some reasonable checks to make sure that
+        # this is a cortex token by checking the JWT keys exist
+        if not payload.get("aud") or not payload.get("sub") or not payload.get("exp"):
+            raise BadTokenException(invalid_token_msg)
         return decodedJWT
-    except py_jwt._JWTError as err:
-        raise BadTokenException(invalidTokenMsg.format(err))
+    except py_jwt._JWTError as err:  # pylint: disable=protected-access
+        raise BadTokenException(invalid_token_msg.format(err)) from err
 
 
-def verify_JWT(token, config = None):
+def verify_JWT(token, config=None):
+    # pylint: disable=invalid-name
     """
     thin wrapper around jwt.decode. This function exists for better error handling of the
     jwt exceptions.
     """
     try:
         decode_JWT(token)
         return token
-    except py_jwt._JWTError:
+    except py_jwt._JWTError:  # pylint: disable=protected-access
         return generate_token(config)
 
 
 def generate_token(config, validity=2):
```

Comment:
Do we need to add getting server time here as an option `use_server_time` ?  like we do in cli to avoid clock drift issues ?

Reply:
will do this as follow on work. This is definitely needed